python/tank/bootstrap/configuration_writer.py

```diff
@@ -547,6 +547,73 @@ def update_roots_file(self, config_descriptor):
             fh.write("\n")
             fh.write("# End of file.\n")
 
+    def is_transaction_pending(self):
+        """
+        Checks if the configuration was previously in the process of being updated but then stopped.
+
+        .. note::
+            Configurations written with previous versions of Toolkit are assumed to completed/
+
+        :returns: True if the configuration was not finished being written on disk, False if it was.
+        """
+
+        # Check if the transaction folder exists...
+        if os.path.exists(
+            self._get_configuration_transaction_folder()
+        ):
+            # ... in which case we'll look for the special file telling us the configuration
+            # is completed.
+            if os.path.exists(self._get_configuration_transaction_filename()):
+                log.debug("Found transactional marker, configuration is complete.")
+                return False
+            else:
+                log.warning("It seems the configuration was not written properly on disk.")
+                return True
+        else:
+            # ... if the folder doesn't exist than we don't even have a config at the moment.
+            log.debug("No transaction folder was found. Assuming valid.")
+            return False
+
+    def start_transaction(self):
+        """
+        Wipes the transaction marker from the configuration.
```

Comment:
this is the bit that i wasn't sure about - i think it's cleaner if we think about the cached configuration as something we can just add to but never remove from - the only time we remove is the backup procedure, which lifs the entire config out and makes way for a new config.

So in this case, maybe the way to do it is to raise an exception in case a transaction is already ongoing rather than deleting the marker? I think this approach will make the implementation more resilient to I/O things, permissions etc. Or at least we are not adding any new places where we need to consider what happens if a file cannot be read/written etc .